hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.mono.state.migration;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.state.common.UniqueTokenId;
+import com.hedera.hapi.node.state.token.Nft;
+import com.hedera.node.app.service.mono.state.merkle.internals.BitPackUtils;
+import com.hedera.node.app.service.mono.state.submerkle.EntityId;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hedera.node.app.service.mono.utils.EntityNumPair;
+import com.hedera.node.app.service.mono.utils.NftNumPair;
+import com.hedera.node.app.service.token.ReadableNftStore;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.jetbrains.annotations.NotNull;
+
+public final class NftStateTranslator {
+
+    @NonNull
+    /**
+     * Converts a {@link com.hedera.node.app.service.mono.state.merkle.MerkleUniqueToken} and to {@link Nft}.
+     * @param merkleUniqueToken the {@link com.hedera.node.app.service.mono.state.merkle.MerkleUniqueToken}
+     * @return the {@link Nft} converted from the {@link com.hedera.node.app.service.mono.state.merkle.MerkleUniqueToken}
+     */
+    public static Nft nftFromMerkleUniqueToken(
+            @NonNull final com.hedera.node.app.service.mono.state.merkle.MerkleUniqueToken merkleUniqueToken) {
+        requireNonNull(merkleUniqueToken);
+        var builder = Nft.newBuilder();
+        final var nftIdPair = merkleUniqueToken.getKey().asNftNumPair();
+        builder.id(merkelUniqueTokenToUniqueTokenId(nftIdPair));
+
+        builder.ownerId(AccountID.newBuilder()
+                        .accountNum(
+                                merkleUniqueToken.getOwner().toGrpcAccountId().getAccountNum())
```

Comment:
If we're converting GRPC account IDs, we should convert them all the way to hapi AccountID before starting the builder, and use the full account ID, not just the number.  This code currently throws out the shard and realm.

Reply:
correct there is another task for @iwsimon and @kimbor to add all as part of the process of the protobuf conversion from AccountId and TokenId

Reply:
Why defer?  These two lines are pretty easy to update.